lms/djangoapps/email_marketing/signals.py

```diff
@@ -54,41 +55,33 @@ def add_email_marketing_cookies(sender, response=None, user=None,
         if sailthru_content:
             post_parms['cookies'] = {'anonymous_interest': sailthru_content}
 
-    try:
-        sailthru_client = SailthruClient(email_config.sailthru_key, email_config.sailthru_secret)
-        log.info(
-            'Sending to Sailthru the user interest cookie [%s] for user [%s]',
-            post_parms.get('cookies', ''),
-            user.email
-        )
-        time_before_call = datetime.datetime.now()
+    time_before_call = datetime.datetime.now()
+    sailthru_response = get_email_cookies_via_sailthru.delay(user.email, post_parms)
 
-        sailthru_response = \
-            sailthru_client.api_post("user", post_parms)
+    try:
+        # synchronous call to celery task with a timeout limit
+        sailthru_response.get(timeout=email_config.user_registration_cookie_timeout_delay,
+                              propagate=True)
+        cookie = sailthru_response.result
 
+    except TimeoutError as exc:
+        log.error("Timeout error while attempting to obtain cookie from Sailthru: %s", unicode(exc))
```

Comment:
> Docker devstack is single-process just like vagrant devstack, so the failures there are not unexpected.

To be clear here, the issue of synchronization/timeout i am getting was on my own setup of rabbitmq/celery outside of docker to test the use case i wanted.  I still don't know for now how to run/switch docker for different setting like devstack_with_worker or is it possible. It seems docker runs with CELERY_ALWAYS_EAGER=True, which is not i want.  Ok I'll try to manually test it via browser.